packages/modules/api-key/src/joiner-config.ts

```diff
@@ -1,31 +1,11 @@
 import { Modules } from "@medusajs/modules-sdk"
-import { ModuleJoinerConfig } from "@medusajs/types"
-import { MapToConfig } from "@medusajs/utils"
-import ApiKey from "./models/api-key"
+import {
+  buildEntitiesNameToLinkableKeysMap,
+  defineJoinerConfig,
+  MapToConfig,
+} from "@medusajs/utils"
 
-export const LinkableKeys: Record<string, string> = {
-  api_key_id: ApiKey.name,
-}
+export const joinerConfig = defineJoinerConfig(Modules.API_KEY)
 
-const entityLinkableKeysMap: MapToConfig = {}
-Object.entries(LinkableKeys).forEach(([key, value]) => {
-  entityLinkableKeysMap[value] ??= []
-  entityLinkableKeysMap[value].push({
-    mapTo: key,
-    valueFrom: key.split("_").pop()!,
-  })
-})
-
-export const entityNameToLinkableKeysMap: MapToConfig = entityLinkableKeysMap
-
-export const joinerConfig: ModuleJoinerConfig = {
-  serviceName: Modules.API_KEY,
-  primaryKeys: ["id"],
-  linkableKeys: LinkableKeys,
-  alias: [
-    {
-      name: ["api_key", "api_keys"],
-      args: { entity: ApiKey.name },
-    },
-  ],
-} as ModuleJoinerConfig
+export const entityNameToLinkableKeysMap: MapToConfig =
```

Comment:
thought: Can we automagically export the joiner config? It would be great to not require an explicit export in the module. Similar to the migration helpers.

Reply:
or if you are talking about what you mention about handling it under the hood, then it wont be part of this pr yet, it requires a bit more work. we still need to discuss some point and I prefer to not extend the scope of the main pr

Reply:
> handling it under the hood

Yeah, this is what I had in mind. Totally fine to save this for a follow-up PR. I was just curious to hear if this was in the pipeline of improvements.

Reply:
it is in the pipeline but not this one, since it will touch the internal loading mechanism I prefer to keep it separated after we merged this one, do we have a ticket for it or should I create on in and in which project?

Reply:
I'll create one 👍